tests/cephadm/test_cephadm_bootstrap_with_spec_file.py

```diff
@@ -0,0 +1,109 @@
+import yaml
+
+from cli.exceptions import ConfigError, ResourceNotFoundError
+from cli.ops.cephadm import bootstrap, set_container_image_config
+from cli.utilities.configure import (
+    get_tools_repo,
+    setup_client_node,
+    setup_installer_node,
+)
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def create_spec_file(installer, nodes, specs, spec_file):
+    """create spec file"""
+    spec_file = installer.remote_file(
+        sudo=True, file_name=f"{spec_file}", file_mode="wb"
+    )
+    for spec in specs:
+        if spec["service_type"] == "host":
+            hostname = spec["hostname"][-1]
+            spec["addr"] = nodes[int(hostname) - 1].ip_address
+            spec["hostname"] = nodes[int(hostname) - 1].hostname
+        else:
+            spec["placement"]["hosts"] = [
+                nodes[int(node[-1]) - 1].hostname for node in spec["placement"]["hosts"]
+            ]
+        yaml.dump(spec, spec_file, default_flow_style=False)
+        spec_file.write("---\n")
+
+
+def run(ceph_cluster, **kwargs):
+    """Verify bootstrap a cluster along with a spec file"""
+    # Get config specs
+    config, custom_config = (
+        kwargs.get("config"),
+        kwargs.get("test_data", {}).get("custom-config", {}),
+    )
+
+    # Check mandatory parameter bootstrap
+    if not config.get("bootstrap"):
+        raise ConfigError("Mandatory config 'bootstrap' not provided")
+
+    # Get cluster config
+    installer, nodes = (
+        ceph_cluster.get_nodes(role="installer")[0],
+        ceph_cluster.get_nodes(),
+    )
+
+    # Get build details
+    rhbuild, build_type, ibm_build, tools_repo, image = (
+        config.get("rhbuild"),
+        config.get("build_type"),
+        config.get("ibm_build", False),
+        config.get("base_url"),
+        config.get("container_image"),
+    )
+
+    # Get tools repo url
+    tools_repo = get_tools_repo(tools_repo, ibm_build)
+
+    # Get ansible preflight tag
+    ansible_preflight = config.get("ansible_preflight")
+
+    # Configure installer node
+    setup_installer_node(
+        installer, nodes, rhbuild, tools_repo, build_type, ibm_build, ansible_preflight
+    )
+
+    # Get bootstrap config
+    bootstrap_config = config.get("bootstrap")
+
+    # create spec file
+    create_spec_file(
```

Comment:
as this is part of bootstrap, move this method to - https://github.com/red-hat-storage/cephci/blob/master/cli/ops/cephadm.py#L141